addons/udes_stock_refactoring/models/stock_move.py

```diff
@@ -359,6 +359,132 @@ def _refactor_action_by_maximum_quantity(self, maximum_qty):
+            remaining_moves -= moves_used
+            grouped_moves.append(moves_used)
+        return grouped_moves
+
+    def _moves_for_weight(self, weight, sort=True):
+        """
+        Return a subset of moves from self where their sum of the weight
+        to do is equal to parameter weight.
+        In case that a move needs to be split, the new move is
+        also returned (this happens when total weight * quantity in the move is
+        greater than weight parameter).
+        If there is not enough to do in the moves,
+        also return the remaining weight.
+        """
+        StockMove = self.env["stock.move"]
+        new_move = None
+        result = StockMove.browse()
+        if weight == 0:
+            return result, new_move, weight
+
+        if sort:
+            sorted_moves = self.sorted(
+                lambda m: m.product_id.weight * m.product_uom_qty, reverse=True
+            )
+            greater_equal_moves = sorted_moves.filtered(
+                lambda m: float_compare(
+                    (m.product_id.weight * m.product_uom_qty),
+                    weight,
+                    precision_rounding=m.product_uom.rounding,
+                )
+                >= 0
+            )
+            # last one will be at least equal
+            moves = greater_equal_moves[-1] if greater_equal_moves else sorted_moves
+        else:
+            moves = self
+
+        for move in moves:
+            result |= move
+            extra_weight = (move.product_id.weight * move.product_uom_qty) - weight
+            if extra_weight > 0:
+                if move.product_id.weight > weight:
+                    # In cases where the product weight exceeds the weight,
+                    # split so a single qty remains
+                    extra_qty = move.product_uom_qty - 1
+                else:
+                    extra_qty = extra_weight / move.product_id.weight
+                # Raise if extra_qty is not supported by the uom. This should never happen,
+                # but it would be better to raise than to end up with bad qtys of products to be picked
+                rounded_uom_amount = tools.float_round(
+                    extra_qty,
+                    precision_rounding=move.product_id.uom_id.rounding,
+                    rounding_method="UP",
+                )
+                if rounded_uom_amount != extra_qty:
+                    raise ValidationError(
+                        _(
+                            "Refactor by weight attempted to split a move in a way which the uom (%s) of the product does not support. "
+                            "Wanted to split off %s, the uom would only support splitting off %s"
+                        )
+                        % (move.product_id.uom_id.name, extra_qty, rounded_uom_amount)
+                    )
+                # TODO palabaster: This is post confirm action, so why tf do I need to do this to avoid the UserError? :(
```

Comment:
Do we want to resolve this?
What happens if it is triggered on action assign?

Reply:
Oops, i forgot about this one! I'll look into it now

Reply:
519a5f3a0d8900a9d687b5cb27f590c1169b3b5b

Reply:
>  What happens if it is triggered on action assign?
I've smoke tested both when Pick is configured to be post assign, and post confirm, and both seem to work as expected